python/lsst/summit/utils/guiders/reading.py

```diff
@@ -90,6 +90,8 @@ class GuiderData:
         Camera rotation angle in degrees.
     isMedianSubtracted : `bool`, optional
         If True, the raw stamps have been subtracted.
+    columnMaskK : `float`, optional
+        Threshold factor used for column mask detection (default 30.0).
```

Comment:
Please change at least these two, and any others you can, to being immutable properties - it shouldn't be possible to change these downstream as they're properties which are set when the object is created.

Reply:
They should therefore also not be optional/have default values at all.

Reply:
Got it, i had to make pydantic class to make them imutable.